File: notes_staging/extract.js
const fs = require('fs');
const path = require('path');
const pdfParse = require('pdf-parse');

async function extractAll() {
    const files = fs.readdirSync(__dirname).filter(f => f.endsWith('.pdf'));
    let combinedText = '';

    for (const file of files) {
        console.log(`Extracting ${file}...`);
        const dataBuffer = fs.readFileSync(path.join(__dirname, file));
        try {
            const pdfParseModule = await import('pdf-parse');
            const parser = pdfParseModule.default ? pdfParseModule.default : pdfParseModule;
            const data = await parser(dataBuffer);
            combinedText += `\n\n--- Start of ${file} ---\n\n`;
            combinedText += data.text;
        } catch(err) {
            console.error(`Failed to parse ${file}`, err);
        }
    }

    fs.writeFileSync(path.join(__dirname, 'extracted_notes.txt'), combinedText);
    console.log('Done extracting to extracted_notes.txt');
}

extractAll();

File: notes_staging/extract.py
import os
from PyPDF2 import PdfReader

def extract_pdfs():
    directory = '.'
    combined_text = ""
    for filename in os.listdir(directory):
        if filename.endswith(".pdf"):
            print(f"Extracting {filename}...")
            try:
                reader = PdfReader(os.path.join(directory, filename))
                combined_text += f"\n\n--- Start of {filename} ---\n\n"
                for page in reader.pages:
                    combined_text += page.extract_text() + "\n"
            except Exception as e:
                print(f"Failed to read {filename}: {e}")

    with open("extracted_notes.txt", "w", encoding="utf-8") as f:
        f.write(combined_text)
    print("Done extracting to extracted_notes.txt")

if __name__ == "__main__":
    extract_pdfs()
